client/src/app/plugins/PluginsRoot.js

```diff
@@ -90,6 +90,7 @@ export default class PluginsRoot extends PureComponent {
             subscribe={ subscribe }
             log={ this.log }
             displayNotification={ this.displayNotification }
+            newFileItems={ app._getNewFileItems() }
```

Comment:
I'm not a big fan of this, but as we do it essentially in the same manner I'd be O.K. merging it. If we decide to do it, let's prefix the new prop with an `_` to denote it is internal.

What I'm wondering: Why not simply inject the `app` (or an API wrapper around it) via  props or the context? We'd be able to get rid of nearly _all_ of these obscure props and rather rely on an interface plug-ins can interact with.

Reply:
Good point! I added a wrapper function calling [`_getFromApp`](https://github.com/camunda/camunda-modeler/pull/2616/commits/5cfe0bd391a50e45072a2539598121c329acef4b#diff-4aea037d3994f2e5ba5ddb32f9eb59573aefba691c4a70b9870fd6a3584a8cccR95) to make this much easier. I'd propose to keep it simple for now and extend it later (e.g. by specifying things that we don't want to expose). I'd also keep other plugin props untouched since we would break some other plugins I guess.